src/Traits/MockTrait.php

```diff
@@ -134,21 +134,43 @@ protected function assertArguments(
         $expectedCount = count($expected);
         $actualCount = count($actual);
 
-        if ($expectedCount !== $actualCount) {
+        if ($isClass) {
+            $reflectionMethod = new \ReflectionMethod($class, $function);
+            $parameters = $reflectionMethod->getParameters();
+            $requiredParametersCount = count(array_filter($parameters, fn($param) => !$param->isOptional()));
+        } else {
+            $parameters = $actual;
             $requiredParametersCount = count(array_filter($actual, fn ($item) => $item !== self::OPTIONAL_ARGUMENT_NAME));
+        }
```

Comment:
let's move it to the separate method which will return [$parameters, $requiredParametersCount]